apps/managedidentity/managedidentity.go

```diff
@@ -204,17 +209,36 @@ func New(id ID, options ...ClientOption) (Client, error) {
 		return Client{}, err
 	}
 
-	// If source is Azure Arc return an error, as Azure Arc allow accepts System Assigned managed identities.
-	if source == AzureArc {
+	// Check for user-assigned restrictions based on the source
+	switch source {
+	case AzureArc:
+		switch id.(type) {
+		case UserAssignedClientID, UserAssignedResourceID, UserAssignedObjectID:
+			return Client{}, errors.New("Azure Arc doesn't support user-assigned managed identities")
+		}
+	case CloudShell:
 		switch id.(type) {
 		case UserAssignedClientID, UserAssignedResourceID, UserAssignedObjectID:
-			return Client{}, errors.New("azure Arc doesn't support user assigned managed identities")
+			return Client{}, errors.New("Cloud Shell doesn't support user-assigned managed identities")
+		}
+	case ServiceFabric:
+		switch id.(type) {
+		case UserAssignedClientID, UserAssignedResourceID, UserAssignedObjectID:
+			return Client{}, errors.New("Service Fabric API doesn't support specifying a user-assigned identity. The identity is determined by cluster resource configuration. See https://aka.ms/servicefabricmi")
 		}
 	}
+
 	opts := ClientOptions{
 		httpClient:         shared.DefaultClient,
 		retryPolicyEnabled: true,
 	}
+	if source == ServiceFabric {
+		h, err := NewHTTPClientWithCustomCertValidation()
```

Comment:
Pls add a comment explaining why we do this.

Reply:
Adding comment